### PR TITLE
Mixing axis boolean filters with other filters 246

### PR DESCRIPTION
**Pull request by alixdamman**
This is just a start. I would like your review before to proceed further.

I can do 
```python
arr = ndtest((4, 10)).set_labels('b', '0..9')
arr['a0', arr.b < 5]
```

### CODE REVIEW

**Comment by gdementen** [comment]
this is awful to review (because of the joined lines). I already asked you not to do that for lines that are not modified for another reason. We should do it all at once (see #94).

**Comment by gdementen** [comment]
Aside from the rewrapping, this seems like a good direction. 

It should be generalized to handle ND boolean filters, eg:

```python
>>> arr = ndtest((3, 6, 4)).set_labels('c', '0..3').set_labels('a', '0..2')
>>> arr[arr.a >= arr.c]
>>> # works
>>> arr[arr.a >= arr.c]['b1:b3']
>>> # works
>>> arr['b1:b3'][arr.a >= arr.c]

>>> # failed (unsure it works with your changes)
>>> arr['b1:b3', arr.a >= arr.c]
>>> # failed (unsure it works with your changes)
>>> arr[arr.a >= arr.c, 'b1:b3']
```

**Comment by alixdamman** [comment]
I removed the commit with rewrapping

**Comment by alixdamman** [comment]
I will test with ND boolean filters.

File: larray/core/array.py [hunk 1]
```diff
@@ -1840,8 +1840,7 @@ def _translate_axis_key_chunk(self, axis_key, bool_passthrough=True):
             raise ValueError("%s is not a valid label for any axis"
                              % axis_key)
         elif len(valid_axes) > 1:
-            # FIXME: .id
-            valid_axes = ', '.join(str(a.id) for a in valid_axes)
+            valid_axes = ', '.join(str(a) for a in valid_axes)
```
Comment (gdementen) [inline larray/core/array.py]: this is not correct for anonymous axes (and part of the reason I never did the change yet). Now that I think of it, it is not very hard either. Something like:
```python
', '.join(axis.name if axis.name is not None else str(self.axes.index(axis)) for axis in valid_axes)
```
should do the trick.

File: doc/source/changes/version_0_25_1.rst.inc [hunk 2]
```diff
@@ -37,3 +37,18 @@ Fixes
 
     >>> arr['a2:'] = arr[':a1'].set_labels('a', 'a2,a3')
     >>> arr['a2:'] = arr[':a1'].drop_labels('a')
+
+* fixed getting a subset of an array by mixing boolean filters and other filters (closes :issue:`246`):
+
+  >>> arr = ndtest((4, 6)).set_labels('b', '0..5')
```
Comment (gdementen) [inline doc/source/changes/version_0_25_1.rst.inc]: * indentation is wrong (should be 4)
* I would use ndrange('a=0..2,b=0..3') and x.b < 2 to make the test shorter

File: doc/source/changes/version_0_25_1.rst.inc [hunk 3]
```diff
@@ -37,3 +37,18 @@ Fixes
...
+  >>> arr
+  a\b   0   1   2   3   4   5
+   a0   0   1   2   3   4   5
+   a1   6   7   8   9  10  11
+   a2  12  13  14  15  16  17
+   a3  18  19  20  21  22  23
+  >>> # equivalent to arr['a0,a2', x.b < 3]
```
Comment (gdementen) [inline doc/source/changes/version_0_25_1.rst.inc]: Unsure showing the equivalence is worth it here

**Comment by gdementen** [comment]
Thanks for removing the rewrap...

**Comment by alixdamman** [comment]
Well,

```python
>>> arr['b1:b3', arr.a >= arr.c]
>>> # or
>>> arr[arr.a >= arr.c, 'b1:b3']
``` 
does not crash but does not give expected result either.

File: larray/core/array.py [hunk 4]
```diff
@@ -1854,6 +1853,12 @@ def _translate_axis_key(self, axis_key, bool_passthrough=True):
         PGroup
             Positional group with valid axes (from self.axes)
         """
+        if isinstance(axis_key, ExprNode):
+            axis_key = axis_key.evaluate(self.axes)
+
+        if isinstance(axis_key, LArray) and np.issubdtype(axis_key.dtype, np.bool_) and bool_passthrough:
+            return PGroup(axis_key.nonzero()[0], axis=axis_key.axes[0])
```
Comment (gdementen) [inline larray/core/array.py]: nonzero()[0] is probably the reason ND boolean filters fail

**Pull request merged**